pool/app/message_template/repo/repo_sql.ml

```diff
@@ -115,29 +116,95 @@ let find_by_label pool ?entity_uuids language label =
           ([], dyn)
           entity_uuids
       in
-      id_list
-      |> CCString.concat ","
-      |> fun ids ->
+      let ids = id_list |> CCString.concat "," in
+      let order_by =
+        Format.asprintf
+          {|
+        FIELD(entity_uuid, %s) DESC,
+        %s
+        |}
+          ids
+          order_by_lang
+      in
       ( Format.asprintf
           {sql| AND(
-                  pool_message_templates.entity_uuid IS NULL
-                  OR
-                  pool_message_templates.entity_uuid IN (%s))
-                ORDER BY
-                FIELD(entity_uuid, %s) DESC
-                LIMIT 1
-              |sql}
-          ids
+              pool_message_templates.entity_uuid IS NULL
+              OR
+              pool_message_templates.entity_uuid IN (%s))
+          |sql}
           ids
         |> where
+      , order_by
       , dyn )
   in
   let (Dynparam.Pack (pt, pv)) = dyn in
   let request =
-    Format.asprintf "%s WHERE %s" select_sql where |> pt ->! RepoEntity.t
+    Format.asprintf "%s WHERE %s ORDER BY %s LIMIT 1" select_sql where order_by
+    |> pt ->! RepoEntity.t
   in
   Utils.Database.find_opt (pool |> Database.Label.value) request pv
   ||> CCOption.to_result Pool_common.Message.(NotFound Field.Template)
+  >|+ fun ({ language; _ } as t) -> t, language
+;;
+
+let find_all_by_entity_uuid_and_label_request dyn languages entity_uuid =
```

Comment:
Our queries are being built in complex ways by now, it might be worth looking into abstracting away raw SQL with a type-safe SQL builder library (or write a layer ourselves). This will improve maintainability for the future. (cc: @mabiede @asaurer)